### PR TITLE
feat: disable guest row in membership table

### LINKED ISSUE #1478: Disable item login schema and show disabled guests

### PR DESCRIPTION
**Pull request by pyphilia**
Included status in ItemLoginSchema, so guest's memberships are disabled when we remove item login.

https://github.com/user-attachments/assets/e39d3651-63ad-4a37-8b56-52a2198ac0d4

close #1478

### CODE REVIEW

File: src/components/item/sharing/membershipTable/GuestItemMembershipTableRow.tsx [hunk 1]
```diff
@@ -1,36 +1,78 @@
...
+  const { data: currentAccount } = hooks.useCurrentMember();
+
+  let isDisabled = false;
+  if (
+    currentAccount?.id !== data.account.id &&
+    (!itemLoginSchema ||
+      itemLoginSchema.status === ItemLoginSchemaStatus.Disabled)
+  ) {
+    isDisabled = true;
+  }
```
Comment (ReidyT) [inline src/components/item/sharing/membershipTable/GuestItemMembershipTableRow.tsx:40]: I'm not sure that the if add clarity compared to an inline const. Maybe we should use to consts?

```suggestion
  const itemLoginSchemaIsDisabled = !itemLoginSchema || itemLoginSchema.status === ItemLoginSchemaStatus.Disabled;
  const isDisabled = currentAccount?.id !== data.account.id && itemLoginSchemaIsDisabled;
```

File: package.json [hunk 2]
```diff
@@ -20,8 +20,8 @@
     "@emotion/styled": "11.13.0",
     "@graasp/chatbox": "3.3.0",
     "@graasp/map": "1.18.0",
-    "@graasp/query-client": "3.24.0",
-    "@graasp/sdk": "4.29.1",
+    "@graasp/query-client": "github:graasp/graasp-query-client#932-update-item-login-schema-state",
```
Comment (ReidyT) [inline package.json:23]: reminder ^^

File: cypress/support/types.ts [hunk 3]
```diff
@@ -23,7 +24,8 @@ import {
 
 export type ItemForTest = DiscriminatedItem & {
   categories?: ItemCategory[];
-  thumbnails?: string;
+  // TODO: INCORRECT! Fix in coming
```
Comment (ReidyT) [inline cypress/support/types.ts:27]: Remove this when the backend is merged 😆

**Review by ReidyT — APPROVED** [review]
Nice changes, LGTM 🐦‍🔥 I just add a suggestion about the way we check `isDisabled`.

File: cypress/support/types.ts (hunk 3 above)
Comment (pyphilia) [reply cypress/support/types.ts:27]: Actually I will need to merge this before your packed thumbnails, so if it's okay for you I will let that and you will rebase and remove this^^

Comment (ReidyT) [reply cypress/support/types.ts:27]: okay no worries!

**Pull request merged**